Lib/unittest/test/test_async_case.py

```diff
@@ -167,6 +167,25 @@ async def on_cleanup(self):
         test.run()
         self.assertEqual(events, ['asyncSetUp', 'test', 'asyncTearDown', 'cleanup'])
 
+    def test_deprecation_of_return_val_from_test(self):
+        # Issue 41322 - deprecate return of value!=None from a test
+        class Test(unittest.IsolatedAsyncioTestCase):
+            async def test(self):
+                return 1
+            async def test2(self):
+                yield 1
+
+        with self.assertWarns(DeprecationWarning) as w:
+            Test('test').run()
+            self.assertIn('It is deprecated to return a value!=None', w.warnings[0].message.args[0])
+            self.assertIn('test', w.warnings[0].message.args[0])
```

Comment:
'test' is too general name. This substring is a part of the warning message ("... from a **test** case ..."), so the test is passed in any case. Use more specific names, like test1, test2, test_func, test_gen.

Also, would not be better to test `str(w.warnings[0].message)` instead of `w.warnings[0].message.args[0]`?